src/ucp/api/ucp.h

```diff
@@ -1629,11 +1629,11 @@ void ucp_ep_print_info(ucp_ep_h ep, FILE *stream);
 /**
  * @ingroup UCP_ENDPOINT
  *
- * @brief Non-blocking flush of outstanding AMO and RMA operations on the
+ * @brief Non-blocking flush of outstanding SEND, AMO and RMA operations on the
```

Comment:
it is not defined that the flush of SEND means. Do you mean tag send or stream send here?

Reply:
Both, do prefer to document all types of API? We will have AM soon, so it also should be added because it will be flushed as well...

Reply:
but what exactly is the meaning for SEND? for RMA/ATOMIC, it means the remote memory is updated and operation is completed. for SEND, we can't always guarantee remote completion, because it depends on matching (e.g RNDV)

Reply:
AFAIK flush (it's current implementation) doesn't guarantee that the remote memory is updated by RMA put operation

Reply:
according to out API definition, it does (even if in some rare scenario it does not work)
anyway, still not clear, what flushing of SEND means?

Reply:
you know it :) local completion of all outstanding TX operations, pls help with better word if you can but my opinion - current mentioning only AMO and RMA is not correct. flush is widely used for send operations too, for example, before closure.

Reply:
where is it used before closure? (except gtest)?

Reply:
ok then, missed it

Reply:
Ok, at least it mentioned in our documentation
https://github.com/openucx/ucx/blob/a6a6d4c1c46fe6125a0d876377d13088dd8f9e7b/src/ucp/api/ucp.h#L261

Reply:
it's different, ucp_ep_close() can potentially cancel protocols, but regular ep_flush cannot